cadc-vos-server/src/main/java/org/opencadc/vospace/server/pkg/VospacePackageRunner.java

```diff
@@ -237,35 +281,40 @@ protected URL getURL(VOSURI nodeURI)
     }
 
     // Get a PackageItem for a directory from the given ContainerNode.
-    protected PackageItem getDirectoryPackageItem(ContainerNode node) {
-        String nodePath = Utils.getPath(node);
-        log.debug("ContainerNode path: " + nodePath);
-
-        return new PackageItem(nodePath);
+    protected PackageItem getDirectoryPackageItem(ContainerNode node, String relativePath) {
+        return new PackageItem(relativePath);
     }
 
     // Get a PackageItem for a file from the given DataNode.
-    protected PackageItem getFilePackageItem(DataNode node)
+    protected PackageItem getFilePackageItem(DataNode node, String relativePath)
```

Comment:
in getDirectoryPackageItem the ContainerNode arg is ignored and only the relativePath is used, but here in getFilePackageItem it uses both... inconsistent usage style leads to bugs.

It looks like the second arg should be named `parentPath` and probably swapped to be first. In that case getDirectoryPackageItem is the one that's wrong (and maybe how it is called is wrong -- 2 bugs that counter each other?)

Reply:
The relative path is consistent between all 3, so it should be the first arg. For data and link nodes the relative path is the path to the parent node so calling it the parent node makes better sense. For container nodes, it's the path to the node itself. Code comments should make that more clear.

Reply:
why is the meaning of the relative path different for container nodes? it seems like a sign that something else is not right as well... it might work, but will look like a bug and be easy to break if inconsistent